js/client-teambuilder.js

```diff
@@ -393,8 +399,14 @@
 					if (filterFormat && filterFormat !== (team.format || 'gen7')) continue;
 					if (filterFolder !== undefined && filterFolder !== team.folder) continue;
 
-					if (this.curSearchVal !== '' && team.team.indexOf(this.curSearchVal) == -1) {
-						continue;
+					if (this.curSearchVal) {
+						var pokemon = team.team.split(']').map(function (el) {
```

Comment:
Let's add a comment here explaining why we're doing this? Something like:

"If a Pokemon hasn't been given a nickname, species is omitted from the packed team.team in favor of the name field since the name defaults to the species' display name. While eliminating this redundancy between name and species helps with packed team size, the display name unfortunately won't match the ID search term and so we need to special case searching for Pokemon here"